static/app/components/charts/worldMapChart.tsx

```diff
@@ -46,136 +37,118 @@ const DEFAULT_CENTER_X = 10.97;
 const DISCOVER_QUERY_LIST_CENTER_Y = -12;
 const DEFAULT_CENTER_Y = 9.71;
 
-class WorldMapChart extends React.Component<Props, State> {
-  state: State = {
-    countryToCodeMap: null,
-    map: null,
-    codeToCountryMap: null,
-  };
+export function WorldMapChart({
+  series,
+  seriesOptions,
+  fromDiscover,
+  fromDiscoverQueryList,
+  ...props
+}: WorldMapChartProps) {
+  const map = worldMap;
+  const codeToCountryMap = Object.fromEntries(
+    Object.entries(countryToCodeMap).map(([country, code]) => [code, country])
+  );
 
-  async componentDidMount() {
-    const [countryToCodeMap, worldMap] = await Promise.all([
-      import('sentry/data/countryCodesMap'),
-      import('sentry/data/world.json'),
-    ]);
-
-    // Echarts not available in tests
-    echarts.registerMap?.('sentryWorld', worldMap.default as any);
-
-    // eslint-disable-next-line
-    this.setState({
-      countryToCodeMap: countryToCodeMap.default,
-      map: worldMap.default,
-      codeToCountryMap: Object.fromEntries(
-        Object.entries(countryToCodeMap.default).map(([country, code]) => [code, country])
-      ),
-    });
-  }
+  const theme = useTheme();
 
-  render() {
-    const {countryToCodeMap, map} = this.state;
+  echarts.registerMap?.('sentryWorld', worldMap as any);
```

Comment:
this should probably only be called once. I think we should keep the comment about tests

Reply:
registerMap still looks like its being called once per render